gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java

```diff
@@ -95,6 +102,27 @@
         LiveDataReactiveStreams.fromPublisher(
             deleteFeatureRequests.switchMapSingle(
                 __ -> deleteActiveFeature().toSingleDefault(true).onErrorReturnItem(false)));
+
+    updateFeature =
+        LiveDataReactiveStreams.fromPublisher(
+            updateFeatureRequests.switchMapSingle(
+                map -> updateFeaturePosition(map).toSingleDefault(true).onErrorReturnItem(false)));
+  }
+
+  public void updateFeature(Feature feature, Point point) {
+    HashMap<Feature, Point> map = new HashMap<>();
+    map.put(feature, point);
+    updateFeatureRequests.onNext(map);
+  }
+
+  public LiveData<Boolean> getUpdateFeature() {
+    return updateFeature;
+  }
+
+  private Completable updateFeaturePosition(Map<Feature, Point> map) {
```

Comment:
Why is this a Map and not just two args? If it's because you wanted to avoid a custom class to bind Feature and Point, you can just use the Feature with the updated position.